ruby-service/app/services/authentication_service.rb

```
@@ -0,0 +1,87 @@
+    return false unless verification[:valid]
+
+    user_id = verification[:user_id]
+    user = @db_handler.find_user_by_id(user_id)
+
+    return false unless user
+
+    case required_role
+    when 'admin'
+      user['role'] == 'admin'
+    when 'user'
+      ['admin', 'user'].include?(user['role'])
```

RuboCop (1.88.2)

[convention] Use `%w` or `%W` for an array of words.

(Style/WordArray)